### PR TITLE
Python Lab version history: Allow teacher to view student's old versions

### PR DESCRIPTION
**Pull request by molly-moen**
When a teacher is viewing a student's project, we want to allow them to view, but not restore, that student's previous versions. Previously the version history button was disabled when viewing a student's project. The button is still disabled when looking at a shared project link.

While working on this, I noticed that we weren't actually displaying the student's code when viewing their project (we still saw the teacher's code). I updated `useSource` to correctly load in the student's code.

## Links
- jira ticket: [CT-714](https://codedotorg.atlassian.net/browse/CT-714)

## Testing story
Tested locally.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/components/versionHistory/VersionHistoryButton.tsx [hunk 1]
```diff
@@ -41,6 +41,13 @@ const VersionHistoryButton: React.FunctionComponent<VersionHistoryProps> = ({
   const isViewingOldVersion = useAppSelector(
     state => state.lab2Project.viewingOldVersion
   );
+  const viewAsUserId = useAppSelector(state => state.progress.viewAsUserId);
+
+  // The version history button is generally disabled in read only mode with two exceptions:
+  // if the user is viewing an old version of the project,
+  // or if this is a teacher view a student's project (in which case they can view old versions, but
```
Comment (ebeastlake) [inline apps/src/lab2/views/components/versionHistory/VersionHistoryButton.tsx]: nit: viewing

File: apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx [hunk 2]
```diff
@@ -66,6 +66,10 @@ const VersionHistoryDropdown: React.FunctionComponent<
     state => state.lab2Project.viewingOldVersion
   );
 
+  // If this is a teacher viewing a student's project, we hide the restore button,
+  // but still allow viewing old versions.
+  const viewAsUserId = useAppSelector(state => state.progress.viewAsUserId);
```
Comment (ebeastlake) [inline apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx:72]: Is there historical context on this variable name? `viewAsUserId` = "view as a teacher" is unintuitive to me, though I'm probably missing context :)

**Review by ebeastlake — APPROVED** [review]

File: apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx (hunk 2 above)
Comment (molly-moen) [reply apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx:72]: Yes, we debated the name and this was the best we came up with: https://github.com/code-dot-org/code-dot-org/pull/58269/files/18b375a99d27398a7cdbfb9d61fd9a102dab2636#r1583489705

File: apps/src/lab2/views/components/versionHistory/VersionHistoryButton.tsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/lab2/views/components/versionHistory/VersionHistoryButton.tsx]: updated!

**Pull request merged**